test/integration/replicationcontroller/replicationcontroller_test.go

```diff
@@ -111,10 +111,15 @@ func newMatchingPod(podName, namespace string) *v1.Pod {
 	}
 }
 
-func rmSetup(t *testing.T) (kubeapiservertesting.TearDownFunc, *replication.ReplicationManager, informers.SharedInformerFactory, clientset.Interface) {
+type tContext struct {
+	testing.TB
+	context.Context
+}
+
+func rmSetup(t *testing.T) (tContext, *replication.ReplicationManager, informers.SharedInformerFactory, clientset.Interface) {
```

Comment:
I've thought a bit more about this and now lean towards putting this into `k8s.io/kubernetes/test/utils/ktesting`.

A test then could start with:
```
func TestSomething(t *testing.T) {
    tCtx := ktesting.Init(t)
...
```

In contrast to the current `ktesting.NewTestContext`, that `Init` would always set up a context with cancellation. Long term, we might also add some configurable per-test deadline.

`tCtx` would be an interface that combines `testing.TB`, `context.Context` and adds `Cancel` (= immediately cancel the context, needed by some tests which start goroutines and wait for them to complete) and `Logger` (= return `logr.Logger) (= convenience function, shorter than `klog.FromContext`).

Reply:
If this `ktesting` turns out to be useful, we could move it into a staging repo. Right now, we can only use it internally, but not for unit tests in staging repos.

I would not move this additional API into `klog/v2/ktesting`. That package should remain minimal because of the strict API rules and focus of klog.

Reply:
I've pushed [dc0b653](https://github.com/kubernetes/kubernetes/pull/122148/commits/dc0b653657c0b4ea92a515e3027f7869e38c1ba0) which implements what I had in mind.